src/Chapter2.hs

```diff
@@ -336,7 +350,29 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList startPos endPos l
+  | startPos >= 0 && startPos <= endPos && endPos < len = drop startPos (take (endPos + 1) l)
+  | startPos >= 0 && startPos < endPos && endPos >= len = drop startPos l
+  | otherwise = []
```

Comment:
Interesting solution! 💡 

Due to the way `drop` and `take` functions work, you can simplify conditions a bit in here. Also, you could avoid using `length` right in the start. This is important because taking `length` is time-consuming operation and we always try to avoid this when possible.

As for solution, you can first start with the check on reasonable input. This will save time to check it first and return an empty list if the input is not correct.

Something like this will work:

```haskell
subList startPos endPos l
    | startPos < 0 || endPos < 0 || endPos < startPos = []
    | otherwise = take (endPos - startPos + 1) (drop startPos l)
```